### DISCUSSION TITLE
How to make a tiled GeoJsonLayer?

### DISCUSSION THREAD

**Discussion by zachcoleman** [visgl/deck.gl#9203] [Q&A] [answered]
I want to make a GeoJsonLayer that uses tiled requests (z/x/y) to the backend and utilizes the [Tileset2D](https://deck.gl/docs/api-reference/geo-layers/tile-layer#tileset2d) class with cacheing and what not. Is this possible? 

I'm fairly sure this is possible, but trying to understand the nuance. I think the data for the respective layer might have to be transformed into a single entity, but this could still be very useful for reducing the number of requests to a backend.

**Comment by Pessimistress** [comment, marked as answer]
You are describing exactly what TileLayer does.

**Reply by zachcoleman** [reply]
Oh! I think I understand now.

To elaborate are you saying this is exactly what TileLayer does and I just need to change the sub layer rendering? like [here](https://deck.gl/docs/api-reference/geo-layers/tile-layer#tileset2d:~:text=renderSubLayers%3A%20props%20%3D%3E%20%7B%0A%20%20%20%20const%20%7BboundingBox%7D%20%3D%20props.tile%3B%0A%0A%20%20%20%20return%20new%20BitmapLayer(props%2C%20%7B%0A%20%20%20%20%20%20data%3A%20null%2C%0A%20%20%20%20%20%20image%3A%20props.data%2C%0A%20%20%20%20%20%20bounds%3A%20%5BboundingBox%5B0%5D%5B0%5D%2C%20boundingBox%5B0%5D%5B1%5D%2C%20boundingBox%5B1%5D%5B0%5D%2C%20boundingBox%5B1%5D%5B1%5D%5D%0A%20%20%20%20%7D)%3B%0A%20%20%7D%2C) to something like:

```javascript
renderSubLayers: props => {
    return new GeoJsonLayer(
       id: //<something derived from tile number>,
       data: props.data,
       ...
    );
  },
```
